downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -380,14 +383,17 @@ abstract class VersionControlSystem(
 
     /**
      * Update the [working tree][workingTree] by checking out the given [revision], optionally limited to the given
-     * [path] and [recursively][recursive] updating any nested working trees. Return a [Result] that encapsulates the
-     * originally requested [revision] on success, or the occurred exception on failure.
+     * [path] and [recursively][recursive] updating any nested working trees.
+     * Optionally, a VCS-specific configuration can be supplied via [vcsConfiguration].
+     * Return a [Result] that encapsulates the originally requested [revision] on success,
+     * or the occurred exception on failure.
      */
     abstract fun updateWorkingTree(
         workingTree: WorkingTree,
         revision: String,
         path: String = "",
-        recursive: Boolean = false
+        recursive: Boolean = false,
+        vcsConfiguration: VersionControlSystemConfiguration? = null
```

Comment:
Conceptually, this `vcsConfiguration` should not be passed to a specific function, but to the constructor. IMO the right way to do would be to introduce a `VersionControlSystemFactory` that inherits from `TypedConfigurablePluginFactory`, like e.g. `ScannerWrapperFactory` does.

Reply:
Agreed. I convert the PR back to draft and create a solution based on TypedConfigurablePluginFactory, like e.g. ScannerWrapperFactory does.